tpm2/tpm2.go

```diff
@@ -1461,8 +1484,11 @@ func SignWithSession(rw io.ReadWriter, sessionHandle, key tpmutil.Handle, passwo
 
 // Sign computes a signature for digest using a given loaded key. Signature
 // algorithm depends on the key type.
-func Sign(rw io.ReadWriter, key tpmutil.Handle, password string, digest []byte, sigScheme *SigScheme) (*Signature, error) {
-	return SignWithSession(rw, HandlePasswordSession, key, password, digest, sigScheme)
+// If 'key' references a Restricted Decryption key, 'validation' must be a valid hash verification
+// ticket from the TPM, which can be obtained by using Hash() to hash the data with the TPM.
+// If 'validation' is nil, a NULL ticket is passed to TPM2_Sign.
+func Sign(rw io.ReadWriter, key tpmutil.Handle, password string, digest []byte, sigScheme *SigScheme, validation *Ticket) (*Signature, error) {
```

Comment:
Bikeshed on ordering here, `Hash` returns `(digest []byte, validation *Ticket)`, so should we have `Sign` take the parameters like: `digest []byte, validation *Ticket, sigScheme *SigScheme`

Reply:
SGTM. I had originally had this order, then swapped it to match the TPM spec, but I agree that this order makes more sense